crates/workspace/src/workspace.rs

```diff
@@ -6717,141 +6743,223 @@ mod tests {
         });
     }
 
-    #[gpui::test]
-    async fn test_join_pane_into_next(cx: &mut gpui::TestAppContext) {
```

Comment:
Note for reviewer: 

Very little in this existing test has been changed, I've just moved it into its own `mod`, so the indentation level changed. This PR might be easier to review if you look at the diff with whitespace disabled.

Reply:
But what was the point to move that away into its own mod?
The rest of the tests are just fine without it.

Reply:
I had two thoughts about it: one was that it seems that joining panes is accumulating functionality (the other issue also mentioned having the ability to join in a certain direction). I figured a single container for those tests makes sense. The other reason was that it provided a nice container for utility/setup functions used by these tests.

Happy to undo it if you prefer.

Reply:
I think this is the least of the issues with the PR, so whatever you prefer.
My personal preference would be to have a single `mod test` and all tests inside that though.